### ISSUE TITLE
Current span is lost after finishing one of the nested ones

### ISSUE THREAD

**Issue by vti** [team-at-cpan/OpenTracing#2]
Current span is lost after finishing one of the nested ones. Is this by design?
Here is an example:

```
my $parent = $tracer->span(operation_name => 'parent');

my $span1 = $tracer->span(operation_name => 'child1');
ok( $tracer->current_span);
my $span2 = $tracer->span(operation_name => 'child2');
ok( $tracer->current_span);

$span2->finish;
ok( $tracer->current_span); # it is lost here
```
I've tried to convert current_span to a stack (so current_span is pushed and popped instead of just reassigned) and it's working fine. Would you consider this change? Or accepting the PR?

**Comment by vti** [comment]
Something like this:

```diff
diff --git a/lib/OpenTracing/Tracer.pm b/lib/OpenTracing/Tracer.pm
index 3def285..b0abb69 100644
--- a/lib/OpenTracing/Tracer.pm
+++ b/lib/OpenTracing/Tracer.pm
@@ -41,6 +41,7 @@ use Log::Any qw($log);
 sub new {
     my ($class, %args) = @_;
     $args{span_completion_callbacks} ||= [];
+    $args{current_span} ||= [];
     bless \%args, $class
 }
 
@@ -133,20 +134,21 @@ sub span {
     $self->add_span(
         my $span = OpenTracing::Span->new(
             tracer => $self,
-            parent => $self->{current_span},
+            parent => $self->{current_span}->[-1],
             %args
         )
     );
-    $self->{current_span} = $span;
+    push @{ $self->{current_span} }, $span;
     return OpenTracing::SpanProxy->new(span => $span);
 }
 
-sub current_span { shift->{current_span} }
+sub current_span { shift->{current_span}->[-1] }
 
 sub finish_span {
     my ($self, $span) = @_;
     $log->tracef('Finishing span %s', $span);
-    undef $self->{current_span} if $self->{current_span} and refaddr($self->{current_span}) == refaddr($span);
+
+    @{ $self->{current_span} }  = grep { refaddr($_) != refaddr($span)} @{ $self->{current_span} };
 
     return $span unless $self->is_enabled;
     push @{$self->{finished_spans} //= []}, $span;

```

**Comment by tm604** [comment]
Hopefully this is fixed now that #6 is merged, but I'll work on some test coverage for this as well. Thanks for the report and patch!
